### PR TITLE
fix(css selector): support comma-separated selector lists

### LINKED ISSUE #2117: [REGRESSION]: Cannot query selector all when the selector is a set of elements
**Context:**
- GOOD Playwright Version: [v0.13.0]
- BAD Playwright Version: [v0.17.0]
- Operating System: [all]
- Extra: []

**Code Snippet**

jest test:
```javascript
test('should query all elements', async (): Promise<void> => {
    // Given
    browser = await chromium.launch({ headless: true });
    const browserContext = await browser.newContext({ viewport: null });
    const page = await browserContext.newPage();
    const url = `file:${path.join(__dirname, 'get-handles-with-value.test.html')}`;
    await page.goto(url);

    // When
    const elements = await page.$$('[role="row"] select, [role="row"] input');

    // Then
    expect(elements.length).toBe(4);
  });
```

get-handles-with-value.test.html
```html
<!DOCTYPE html>
<html>

<head>
  <meta charset="utf-8" />

  <title>Query Selector All from</title>
</head>

<body>
  <table border="1">
    <tr role="row">
      <td>
        <select data-test-id="my-select">
          <option value="1" selected>label 1</option>
          <option value="2">label 2</option>
          <option value="3">label 3</option>
        </select>
      </td>
      <td>row1-cell2</td>
      <td></td>
    </tr>
    <tr role="row">
      <td>
        <select data-test-id="my-select">
          <option value="1">label 1</option>
          <option value="2" selected>label 2 row 2</option>
          <option value="3">label 3</option>
        </select>
      </td>
      <td>row2-cell2</td>
      <td><input type="text" value="2" data-e2e="foobar" /></td>
    </tr>
    <tr role="row">
      <td>
        <select data-test-id="my-select">
          <option value="1">label 1</option>
          <option value="2">label 2</option>
          <option value="3" selected>label 3 row3</option>
        </select>
      </td>
      <td>row3-cell2</td>
      <td>cell3</td>
    </tr>
  </table>
</body>

</html>
```

**Describe the bug**

`page.$$(selector)` throws the exception:

```sh
Evaluation failed: DOMException: Failed to execute 'matches' on 'Element': 'select,' is not a valid selector.

      at matches (__playwright_evaluation_script__:328:21)
      at queryShadowAllInternal (__playwright_evaluation_script__:313:35)
      at Object.queryAll (__playwright_evaluation_script__:276:17)
      at SelectorEvaluator.querySelectorAll (__playwright_evaluation_script__:912:59)
      at __playwright_evaluation_script__:2:57
      at __playwright_evaluation_script__:2:101
      at CRExecutionContext.evaluate (node_modules/playwright-core/lib/chromium/crExecutionContext.js:82:23)
        -- ASYNC --
      at Frame.<anonymous> (node_modules/playwright-core/lib/helper.js:63:23)
      at Page.$$ (node_modules/playwright-core/lib/page.js:150:33)
      at Page.<anonymous> (node_modules/playwright-core/lib/helper.js:64:31)
      at Object.<anonymous>
```

### PR DESCRIPTION
**Pull request by dgozman**
Fixes #2117.

### CODE REVIEW

File: src/injected/cssSelectorEngine.ts [hunk 1]
```diff
@@ -87,73 +87,99 @@ export function createCSSEngine(shadow: boolean): SelectorEngine {
       //   return simple;
       // if (!shadow)
       //   return;
-      const parts = split(selector);
-      if (!parts.length)
+      const selectors = split(selector).filter(parts => !!parts.length).map(parts => parts.reverse());
```
Comment (JoelEinbinder) [inline src/injected/cssSelectorEngine.ts:90]: nit: this line looks like it goes inside `split` to me.

File: src/injected/cssSelectorEngine.ts [hunk 2]
```diff
@@ -87,73 +87,99 @@ export function createCSSEngine(shadow: boolean): SelectorEngine {
...
-    const element = matching[i];
-    if (parts.length === 1 || matches(element, parts, boundary))
-      return element;
+function queryShadowInternal(boundary: SelectorRoot, root: SelectorRoot, selectors: string[][], shadow: boolean): Element | undefined {
+  let elements: NodeListOf<Element> | undefined;
+  if (selectors.length === 1) {
+    const parts = selectors[0];
+    const matching = root.querySelectorAll(parts[0]);
+    for (let i = 0; i < matching.length; i++) {
+      const element = matching[i];
```
Comment (JoelEinbinder) [inline src/injected/cssSelectorEngine.ts:117]:
```suggestion
    for (const element of matching) {
```

File: src/injected/cssSelectorEngine.ts [hunk 3]
```diff
@@ -87,73 +87,99 @@ export function createCSSEngine(shadow: boolean): SelectorEngine {
...
+      if (parts.length === 1 || ancestorsMatch(element, parts, boundary))
+        return element;
+    }
+  } else {
+    elements = root.querySelectorAll('*');
+    for (let i = 0; i < elements.length; i++) {
+      const element = elements[i];
```
Comment (JoelEinbinder) [inline src/injected/cssSelectorEngine.ts:124]:
```suggestion
    for (const element of matching) {
```

File: src/injected/cssSelectorEngine.ts [hunk 4]
```diff
@@ -87,73 +87,99 @@ export function createCSSEngine(shadow: boolean): SelectorEngine {
...
-    if (parts.length === 1 || matches(element, parts, boundary))
-      return element;
+function queryShadowInternal(boundary: SelectorRoot, root: SelectorRoot, selectors: string[][], shadow: boolean): Element | undefined {
+  let elements: NodeListOf<Element> | undefined;
+  if (selectors.length === 1) {
+    const parts = selectors[0];
+    const matching = root.querySelectorAll(parts[0]);
+    for (let i = 0; i < matching.length; i++) {
+      const element = matching[i];
+      if (parts.length === 1 || ancestorsMatch(element, parts, boundary))
```
Comment (JoelEinbinder) [inline src/injected/cssSelectorEngine.ts:118]: This is equivalent right? I'm having trouble tracking the special casings of 1 and 0
```suggestion
      if (ancestorsMatch(element, parts, boundary))
```

**Review by JoelEinbinder — APPROVED** [review]

File: src/injected/cssSelectorEngine.ts (hunk 1 above)
Comment (dgozman) [reply src/injected/cssSelectorEngine.ts:90]: Done.

File: src/injected/cssSelectorEngine.ts (hunk 2 above)
Comment (dgozman) [reply src/injected/cssSelectorEngine.ts:117]: Done.

File: src/injected/cssSelectorEngine.ts (hunk 4 above)
Comment (dgozman) [reply src/injected/cssSelectorEngine.ts:118]: Done.

File: src/injected/cssSelectorEngine.ts (hunk 3 above)
Comment (dgozman) [reply src/injected/cssSelectorEngine.ts:124]: Done.

**Pull request merged**